### ISSUE TITLE
aws_ecs_service: diffs didn't match during apply

### ISSUE THREAD

**Issue by jonatanblue** [hashicorp/terraform#7788]
### Overview

When re-running `terraform apply` after changing an `aws_elb` config, the `aws_ecs_services` fail to create, with this error:

```
* aws_ecs_service.sinatra_backend: diffs didn't match during apply. This is a bug with Terraform and should be reported as a GitHub Issue.
```
### Terraform Version

Terraform v0.6.16
### Affected Resource(s)
- aws_ecs_service
- aws_elb
### Terraform Configuration Files

I have deployed the example app from this [infrastructure-as-code-talk repo](https://github.com/jonatanblue/infrastructure-as-code-talk).

The repo creates an AutoScaling Group with five EC2 instances, two ELBs and four ECS tasks (effectively Docker containers). Two tasks are web servers, attached to one ELB, the other two are back-end services, attached to another ELB.

When first created, both ELBs are public facing. After deploying it the first time, I wanted to make the back-end ELB internal (not public facing), so I added one line to the back-end ELB config in the file [sinatra-backend.tf](https://github.com/jonatanblue/infrastructure-as-code-talk/blob/master/terraform-templates/sinatra-backend.tf#L6):

```
internal = true
```
### Debug Output

Here's the output of `terraform plan` and `terraform apply` (run twice).

https://gist.github.com/jonatanblue/47bbe65e335effcba3e187063fe4507b
### Expected Behavior

The ELB should be destroyed and re-created and the ECS tasks recreated.
### Actual Behavior

When running `terraform apply`, the old ELB is deleted and a new created. However, it looks like terraform crashes out when re-creating the tasks.

If I re-run `terraform apply` the back-end tasks (containers) are created and the error is gone.

If I set `internal = true` from the start, there is no problem. If I change it to `internal = false` and run `terraform apply` the same error occurs. Re-running `terraform apply` still works.

However, changing other things (such as `name`) that require destroy/re-create of the ELB does not cause an error.
### Steps to Reproduce

Your IAM user needs to have access to EC2, AutoScaling, ECS and IAM.

```
export AWS_ACCESS_KEY_ID=redacted
export AWS_SECRET_ACCESS_KEY=redacted
git clone https://github.com/jonatanblue/infrastructure-as-code-talk
cd infrastructure-as-code-talk/terraform-templates
git checkout 3b8b15c4381a33494728ad466ec06368b886b984
cp terraform.tfvars.sample terraform.tfvars
```

Edit the file `terraform.tvars` to match your environment with values for `key_pair_name`, `vpc_id`, `elb_subnet_ids` and `ecs_cluster_subnet_ids`.

Apply original config

```
terraform apply   # This should work
```

Set back-end ELB scheme to [internal](https://www.terraform.io/docs/providers/aws/r/elb.html#internal), by adding the line `internal = true` to the backend elb in `sinatra-backend.tf`:

```
...
 resource "aws_elb" "sinatra_backend" {
   name = "sinatra-backend"
+  internal = true
   subnets = ["${split(",", var.elb_subnet_ids)}"]
...
```

Run terraform again:

```
terraform plan    # This should work, reporting changes to be made

terraform apply   # Should give the error `aws_ecs_service.sinatra_backend: diffs didn't match during apply.`

terraform apply   # Should work
```
### References

Possible duplicate: #7642

**Comment by mtougeron-gpsw** [comment]
FWIW, I've found that errors occur for me when the container port changes as well.

**Comment by mtougeron** [comment]
FWIW, I've found that errors occur for me when the container port changes as well.

**Comment by mitchellh** [comment]
Hello! This should be fixed in later versions of Terraform (0.7.11 at time of writing). 

Its hard to know 100% for a couple reasons:

  * This is on a now much-older version of Terraform. This is our fault, not yours, for not trying sooner, but we've fixed many diff mismatch issues since then.

  * We've fixed issues with exact error messages as this in those versions. 

Please try again! If the issue persists please just open a new issue. Thanks!
